src/scipp/format/formatter.py

```diff
@@ -0,0 +1,48 @@
+def format_variable(data, spec):
+    """
+    String formats the Variable according to the provided specification.
+    Parameters
+    ----------
+    obj
+        A scalar or array-like scipp Variable object
+    spec
+        Format specification; only 'c' for Compact error-reporting supported at present
+
+    Returns
+    -------
+    The formatted string
+    """
+    from scipp import Unit
+    dtype = str(data.dtype)
+    if not any([x in dtype for x in ('float', 'int')]) or spec is None or len(spec) < 1:
+        return data.__repr__()
+    compact = spec[-1] == 'c'
+    is_scalar = False if data.shape else True
+    val = data.value if is_scalar else data.values
+    var = data.variance if is_scalar else data.variances
+    unt = "" if data.unit == Unit('dimensionless') else f" {data.unit}"
+    if compact and var is not None:
+        from numpy import floor, log10, round, array, sqrt, power
+        if is_scalar:
+            val = array((val, ))
+            var = array((var, ))
+        err = sqrt(var)
+        p = floor(log10(err))
```

Comment:
This code (digit handling) is pretty opaque. Can you pull this out into its own function to document what is going on?

Reply:
Good idea, done in a new commit